src/gcp_scanner/client/interface_client.py

```diff
@@ -0,0 +1,36 @@
+#  Copyright 2023 Google LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from abc import ABCMeta, abstractmethod
+
+from googleapiclient import discovery
+from httplib2 import Credentials
+
+
+class IClient(metaclass=ABCMeta):
+  """Interface for Client Classes."""
+
+  @staticmethod
+  @abstractmethod
+  def get_service(credentials: Credentials) -> discovery.Resource:
+    """Create a client.
+
+    Args:
+      credentials: An google.oauth2.credentials.Credentials object.
+
+    Returns:
+      An object of discovery.Resource
+    """
+
+    raise NotImplementedError("Child class must implement create_client")
```

Comment:
do you mean "must implement get_service" here?

Reply:
fixed :pray: